state-reconstruct-fetcher/src/database.rs

```diff
@@ -0,0 +1,152 @@
+use std::{convert::Into, ops::Deref, path::PathBuf};
+
+use ethers::types::{U256, U64};
+use eyre::{Report, Result};
+use rocksdb::{Options, DB};
+
+const INDEX_TO_KEY_MAP: &str = "index_to_key_map";
+const KEY_TO_INDEX_MAP: &str = "key_to_index_map";
+const METADATA: &str = "metadata";
+
+const LAST_REPEATED_KEY_INDEX: &str = "LAST_REPEATED_KEY_INDEX";
+/// The latest l1 block number that was processed.
+const LATEST_L1_BLOCK_NUMBER: &str = "LATEST_L1_BLOCK_NUMBER";
+/// The latest l2 block number that was processed.
+const LATEST_L2_BLOCK_NUMBER: &str = "LATEST_L2_BLOCK_NUMBER";
```

Comment:
They do introduce a lot of contention, but keeping yet another data source would complicate usage, with a potential for lots of interesting bugs due to desynchronization... I think the correct solution is to access the db without locking, but I'd be more comfortable implementing that if it worked first... :-) And as you say, it might not even be _that_ inefficient...

Reply:
Yeah, no need to over-complicate this now, we can look into this another time.